File: 20190519python/0804. Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        s_str="abcdefghijklmnopqrstuvwxyz"
        morse_list=[".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        dict_morse_str={}
        for k in range(len(s_str)):
            dict_morse_str[s_str[k]]=morse_list[k]
        
        word_morse_set=set()
        for word in words:
            cur=""
            for s in word:
                cur+=dict_morse_str[s]
            word_morse_set.add(cur)
        
        return len(word_morse_set)

File: archives/C#/0804. Unique Morse Code Words.cs
public class Solution {
    public int UniqueMorseRepresentations(string[] words) {
        IList<string> morseList=new List<string>{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Dictionary<char,string> morseDic=new Dictionary<char,string>();
        string s="abcdefghijklmnopqrstuvwxyz";
        for (int i=0;i<26;i++){
            morseDic[s[i]]=morseList[i];
        }
        
        HashSet<string> wordSet=new HashSet<string>();
        foreach(string word in words){
            string newWord=word.ToLower();
            string rep="";
            foreach(char c in newWord){
                rep+=morseDic[c];
            }
            wordSet.Add(rep);
        }
        return wordSet.Count;
    }
}
